en/components/texthighlight.md

```diff
@@ -0,0 +1,340 @@
+            let count = 0;
+            const matchesArray = [];
+
+            this.highlights.forEach((h) => {
+                count += h.highlight(this.searchText, this.caseSensitive);
+                matchesArray.push(count);
+            });
+
+            this.matchCount = count;
+
+            this.index += increment;
+            this.index = this.index < 0 ? this.matchCount - 1 : this.index;
+            this.index = this.index > this.matchCount - 1 ? 0 : this.index;
+
+            if (this.matchCount) {
+                let row;
+
+                for (let i = 0; i < matchesArray.length; i++) {
+                    if (this.index < matchesArray[i]) {
+                        row = i;
+                        break;
+                    }
+                }
+
+                const actualIndex = row === 0 ? this.index : this.index - matchesArray[row - 1];
+
+                IgxTextHighlightDirective.setActiveHighlight("group1", {
+                    columnIndex: 0,
+                    index: actualIndex,
+                    page: 0,
+                    rowIndex: row
+                });
+            }
+        } else {
+            this.highlights.forEach((h) => {
+                h.clearHighlight();
+            });
+            this.matchCount = 0;
+        }
+    }
+```
+
+<div class="sample-container loading" style="height: 400px; width: 600px;">
+    <iframe id="text-highlight-2" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/text-highlight-2" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<div>
+    <button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="text-highlight-2" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+
+<div class="divider"></div>
+
+### API Summary
+
+The following tables summarize some of the useful TextHighlight directive inputs and methods.
+
+#### Inputs
+
+The following input is available in the **igxTextHighlight** directive:
+
+| Name | Type | Description |
+| :--- | :--- | :--- |
+| `groupName` | string | Defines the group in which the active highlight will be shared. |
+| `row` | number | Defines the row on which the directive is positioned. |
+| `column` | number | Defines the column on which the directive is positioned. |
+| `page` | number | Defines the page on which the directive is positioned. |
+| `value` | any | Defines the underlying value on which we will perform searches. |
+
+<div class="divider"></div>
+
+#### Methods
+
+The following getters are available in the **igxTextHighlight** directive:
+| Name | Return type | Parameters | Description |
+| :--- | :--- | :--- | :--- |
+| `highlight` | number | The text to highlight and optionally whether the search is case sensitve and should we highlight only exact matches | Highlight the searched string (if it exist in the value input) and returns how many times the value contains the searched highlighted string|
```

Comment:
Should be:
"Highlight**S** the searched string (if it exist**S** in the value input) ..."
instead of:
"Highlight the searched string (if it exist in the value input) ..."